helm-org.el

```diff
@@ -58,9 +58,11 @@ NOTE: This will be slow on large org buffers."
   (org-show-entry))
 
 (cl-defun helm-source-org-headings-for-files (filenames
-                                              &optional (min-depth 1) (max-depth 8))
+                                              &optional (min-depth 1) (max-depth 8) (parents nil))
   (helm-build-sync-source "Org Headings"
-    :candidates (helm-org-get-candidates filenames min-depth max-depth)
+    :candidates (helm-org-get-candidates filenames min-depth max-depth parents)
+    :candidate-transformer (if parents 'reverse nil)
+    :after-init-hook 'helm-follow-mode
```

Comment:
alphapapa <notifications@github.com> writes:

> In helm-org.el:
>
>>    (helm-build-sync-source "Org Headings"
>> -    :candidates (helm-org-get-candidates filenames min-depth max-depth)
>> +    :candidates (helm-org-get-candidates filenames min-depth max-depth parents)
>> +    :candidate-transformer (if parents 'reverse nil)
>> +    :after-init-hook 'helm-follow-mode
>
> That works, but it still has the problem I mentioned: the follow
> happens before the preselect,

Did you try using :follow 1 instead of :after-init-hook ?

But anyway, I am not sure it is a good idea to enable follow-mode by
default (not what I want for myself anyway).

-- 
Thierry
https://emacs-helm.github.io/helm/